### DISCUSSION TITLE
Pixi and Maturin

### DISCUSSION THREAD

**Discussion by lucascolley** [prefix-dev/pixi#1796] [General]
Has anyone used Pixi and Maturin together? Is it easy to transition from a Maturin project to using just pixi? Any guidance on how?

I have [a project](https://github.com/lucascolley/pyo3-array) which uses `pixi.toml` for the Python dependencies and `cargo.toml` for the Rust dependencies. But would like to transition if it's not too difficult!

**Comment by ruben-arts** [comment]
We have an example in [`py-rattler`](https://github.com/conda-incubator/rattler/tree/main/py-rattler) Hopeful this helps you.

**Reply by lucascolley** [reply]
Okay so the Rust dependencies stay in `Cargo.toml`? For a pure Rust project https://prefix-dev.github.io/pixi/v0.27.1/tutorials/rust/ seems to say that you just need `pixi.toml`. I'm guessing that changes if you need to build Python extensions?

**Reply by ruben-arts** [reply]
Yes only `conda` packages can go into the `pixi.toml`. Meaning, dynamically linked binaries and libraries. Rust/Cargo works with source code so you'll need to use their own manager. All our tools are built in Rust and we thus always have both a `pixi.toml` and a `Cargo.toml`. Pixi itself is also a pixi project but without python bindings so no `pyproject.toml` yet
